fhd_core/calibration/calfits_read.pro

```diff
@@ -0,0 +1,181 @@
+Function calfits_read,file_path_fits,obs,params,silent=silent,_Extra=extra
+
+
+  IF file_test(file_path_fits) EQ 0 THEN message,"File: "+file_path_fits+" not found! Returning"
+  
+  t_readfits=Systime(1)
+  lun = fxposit(file_path_fits, 0,/readonly)
+  data_array=mrdfits(lun,0,data_header0,/silent)
+  
+  ;*********Read-in header
+  naxis = sxpar(data_header0,'naxis') ;number of data axes
+  data_dims = INTARR(naxis)
+  for naxis_i=0, naxis - 1 do data_dims[naxis_i] = sxpar(data_header0,'naxis'+strtrim((naxis_i+1),2)) ;dimension of each data axis
+  telescope = sxpar(data_header0,'telescop') ;telescope that took the data
+  gain_convention = sxpar(data_header0,'gnconven') ;convention of the gains, either 'divide' or 'multiply'
+  n_time = sxpar(data_header0,'ntimes') ;number of times with calibration data
+  n_freq = sxpar(data_header0,'nfreqs') ;number of frequencies with calibration data
+  n_ant_data = sxpar(data_header0,'nantsdat') ;number of antennas with calibration data
+  n_jones = sxpar(data_header0,'njones') ;number of polarizations/jones with calibration data
+  caltype = sxpar(data_header0,'caltype') ;type of calibration, 'gain' or 'delay'
+  if STRMATCH(caltype, 'delay*', /FOLD_CASE) then message, 'Input delay calibration not supported at this time.'
+  time_integration = sxpar(data_header0,'inttime') ;time integration of the calibration data
+  freq_channel_width = sxpar(data_header0,'chwidth') ;frequency channel width
+  n_ant_tel = sxpar(data_header0,'nantstel') ;number of antennas in the telescope
+  n_spws = sxpar(data_header0,'nspws') ;number of spectral windows
+  if n_spws GT 1 then message, 'More than one spectral window is not supported at this time.'
+  x_orient = sxpar(data_header0,'xorient') ;Orientation of the X dipole.
+  freq_range = sxpar(data_header0,'frqrange') ;Frequency range that the calibration data is valid for
+  time_range = sxpar(data_header0,'tmerange') ;Time range that the calibration data is valid for
+  
+  data_types = STRARR(naxis)
+  for naxis_i=0, naxis - 1  do data_types[naxis_i] = strtrim(sxpar(data_header0,'ctype'+strtrim((naxis_i+1),2)),2) ;type of each data axis
+  ;Find out which axis is which
+  data_index = where(strmatch(data_types, 'Narrays', /FOLD_CASE) EQ 1)
+  ant_index = where(strmatch(data_types, 'antaxis', /FOLD_CASE) EQ 1)
+  freq_index = where(strmatch(data_types, 'freqs', /FOLD_CASE) EQ 1)
+  time_index = where(strmatch(data_types, 'time', /FOLD_CASE) EQ 1)
+  jones_index = where(strmatch(data_types, 'jones', /FOLD_CASE) EQ 1)
+  data_narray = sxpar(data_header0,strjoin('crval'+strtrim((data_index+1),2))) ;real(gain), imag(gain), flags, (optional input flags), quality
```

Comment:
Documentation for calfits calls them "flags" (https://github.com/HERA-Team/pyuvdata/blob/master/docs/references/calfits_memo.pdf). Is this consistent between your "flags" vs. "weights"?

Reply:
I had assumed the convention was the same for calfits as uvfits, but I had not actually read the calfits documentation. For uvfits, the third column is a float that supposedly corresponds to the weights, but typically has been a jumbled mass of values (e.g. number of time steps and frequency channels averaged together, but with the coarse bandpass also applied). Because of that, we've only been able to usefully test whether it is zero or non-zero, and in FHD anything that is non-zero I set to 1.